### ISSUE TITLE
IllegalStateException on Great Barrier Reef

### ISSUE THREAD

**Issue by Bytekeeper** [OpenBW/BWAPI4J#78]
@Jabbo16  mentioned this on Discord, and I was able to reproduce it.
Using the random seed `1540118558` on `(3)Great Barrier Reef.scx`, the following exception occurs:
```
java.lang.IllegalStateException
    at bwem.Graph.computeDistances(Graph.java:726)
    at bwem.Graph.computeChokePointDistances(Graph.java:654)
    at bwem.Graph.computeChokePointDistanceMatrix(Graph.java:559)
    at bwem.map.MapInitializerImpl.initialize(MapInitializerImpl.java:151)
    at bwem.BWEM.initialize(BWEM.java:57)
    at bwem.BWEM.initialize(BWEM.java:45)
    at ecgberht.Ecgberht.onStart(Ecgberht.java:364)
    at org.openbw.bwapi4j.BW.onStart(BW.java:464)
    at org.openbw.bwapi4j.BW.startGame(Native Method)
    at org.openbw.bwapi4j.BW.startGame(BW.java:163)
```

**Comment by adakitesystems** [comment]
Is this exception being thrown every time or just at random times with this seed? I am not able to reproduce the thrown exception using `TestListenerBwem.java`. I have ran the test bot a few times and it successfully initializes BWEM on GBR each time. The SHA-1 checksum of my local GBR map file is `3506e6d942f9721dc99495a141f41c5555e8eab5`.

**Comment by Bytekeeper** [comment]
@adakitesystems I can reproduce it fine enough. I also found the problem. `ChokePoint` and `Area` have implementations of equal and hashcode which confuse the `computeDistances` algorithm. I removed them and can no longer reproduce the problem.
(to be more precise: 
```
                  final boolean removed =
                      toVisit.remove(new Pair<>(((TileImpl) nextTile).getInternalData(), next));
``` in Graph.java line ~760.
It will remove another item if the distance matches and the ChokePoint is deemed `equal` (but it should only remove on identity and not equality).
